lib/util/clone.ts

```diff
@@ -1,5 +1,33 @@
-import safeStringify from 'fast-safe-stringify';
+import is from '@sindresorhus/is';
 
-export function clone<T>(input: T = null): T {
-  return JSON.parse(safeStringify(input));
+export function clone<T>(input: T, refs = new WeakSet<any>()): T {
+  if (is.array(input)) {
+    refs.add(input);
+    const output: unknown = new Array(input.length);
```

Comment:
```suggestion
    const output: unknown[] = new Array(input.length);
```
Shouldn't this an array type?